src/app/views/interactives/InteractivesDelete.jsx

```diff
@@ -1,133 +1,86 @@
-    static propTypes = {
-        params: PropTypes.shape({
-            interactiveId: PropTypes.string,
-        }),
-        resetInteractiveError: PropTypes.func.isRequired,
-        deleteInteractive: PropTypes.func.isRequired,
-        getInteractive: PropTypes.func.isRequired,
-        rootPath: PropTypes.string.isRequired,
-        errors: PropTypes.object,
-    };
-
-    static contextTypes = {
-        router: PropTypes.object,
-    };
-
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            internal_id: "",
-            title: "",
-            label: "",
-            file: null,
-            slug: "",
-            interactiveId: null,
-            published: false,
-        };
-
-        this.handleDelete = this.handleDelete.bind(this);
-        this.handleReturn = this.handleReturn.bind(this);
-    }
-
-    componentDidMount() {
-        const { interactiveId } = this.props.params;
-        this.setState({ interactiveId: interactiveId });
-        this.props.getInteractive(interactiveId);
-    }
-
-    componentWillReceiveProps(nextProps, nextContext) {
-        if (nextProps.successMessage.success) {
-            const rootPath = this.props.rootPath;
-            if (nextProps.successMessage.type === "delete") {
-                this.props.router.push(`${rootPath}/interactives`);
-            }
+export default function InteractivesDelete(props) {
+    const dispatch = useDispatch();
+    const { successMessage, interactive } = useSelector(state => state.interactives);
+    const { rootPath } = useSelector(state => state.state);
+
+    const [internalId, setInternalId] = useState("");
+    const [title, setTitle] = useState("");
+    const [label, setLabel] = useState("");
+    const [slug, setSlug] = useState("");
+    const [interactiveId, setInteractiveId] = useState("");
+    const [published, setPublished] = useState(false);
+
+    useEffect(() => {
+        const { interactiveId } = props.params;
+        setInteractiveId(interactiveId);
+        dispatch(getInteractive(interactiveId));
+    }, []);
+
+    useEffect(() => {
+        if (interactive.metadata) {
+            const { metadata } = interactive;
+            setInternalId(metadata.internal_id);
+            setTitle(metadata.title);
+            setLabel(metadata.label);
+            setSlug(metadata.slug);
+            setPublished(metadata.published);
         }
+    }, [interactive]);
 
-        if (nextProps.interactive.metadata) {
-            const { metadata } = nextProps.interactive;
-            this.state.title = metadata.title;
-            this.state.label = metadata.label;
-            this.state.slug = metadata.slug;
-            this.state.published = metadata.published;
+    useEffect(() => {
+        if (successMessage.success) {
+            if (successMessage.type === "delete") {
+                props.router.push(`${rootPath}/interactives`);
+            }
         }
-    }
+    }, [successMessage]);
```

Comment:
it looks like your dependency is only [successMessage.success] here

Reply:
sorted